optimizer/lib/transformers/BlurImagePlaceholders.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+const sizeOf = require('image-size');
+const jimp = require('jimp');
+
+/**
+ * Adds placeholders for all AMP images and posters that are blurry versions of
+ * the corresponding original source. The blur will be displayed as the
+ * <amp-img> is rendering, and will fade out once the element is loaded.
+ */
+
+class BlurImagePlaceholders {
+  /**
+   * Parses the document to add blurred placedholders in all appropriate
+   * locations.
+   * @param {Tree} tree - a DOM tree.
```

Comment:
Let's remove type declarations since the true type is not accurate. I.e. I think this is actually a TreeAdapter: https://github.com/inikulin/parse5/blob/master/packages/parse5/docs/tree-adapter/interface.md